runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -71,6 +71,38 @@ struct RegexInfo final {
         replacement(replacement_) {}
 };
 
+class pcre2_match_view {
+public:
+  pcre2_match_view() = default;
+
+  pcre2_match_view(std::string_view subject, PCRE2_SIZE* ovector, int32_t ret_code) noexcept
+      : m_subject_data{subject},
+        m_ovector_ptr{ovector},
+        m_num_groups{ret_code} {}
+
+  int32_t size() const noexcept {
+    return m_num_groups;
+  }
+
+  std::optional<std::string_view> get_group(size_t i) const noexcept {
+    kphp::log::assertion(i >= 0 && i < m_num_groups && m_ovector_ptr);
```

Comment:
Check `i >= 0` is always true since i has type `size_t`. Why is an assert used in the case of an invalid i instead of returning nullopt?